kratos/includes/mortar_classes.h

```diff
@@ -489,23 +489,26 @@ class MortarKinematicVariablesWithDerivatives : public MortarKinematicVariables<
  * This data will be used to compute the derivatives.
  * This class includes different information that is used in order to compute the directional derivatives in the mortar contact conditions
  */
-template< unsigned int TDim, unsigned int TNumNodes>
+template< unsigned int TDim, unsigned int TNumNodes, bool TNormalVariation>
```

Comment:
once this is in the core, it should be tested in the core. 
Also you CAN NOT change this as you wish starting from the moment in which it is there.

i am going to accept this for now, but next time you want to change this you must do a PR on its own, just changing this file and the relevant tests.

Reply:
Ok